### DISCUSSION TITLE
[@types/jsdom] does not work with jsdom@20

### DISCUSSION THREAD

**Discussion by PaulKiddle** [DefinitelyTyped/DefinitelyTyped#60999] [Issues with a @types package]
Upon upgrading to jsdom@20 and running my linter, I receive the following message:
```
node_modules/@types/jsdom/base.d.ts:5:10 - error TS2305: Module '"parse5"' has no exported member 'ElementLocation'.

import { ElementLocation } from "parse5";
```
jsdom@20 includes an upgraded dependency of parse5 which is not compatible with the version used by @types/jsdom

[a comment by an automated account is not shown]

**Comment by lmeerma** [comment]
Hi @sandersn @SimenB @andy-ms @BassT
Sorry for linking you directly, but you were one of the latest contributors to the jsdom type defintions. Can someone maybe take a look at this problem? Thank you!

See also: #60969

**Reply by BassT** [reply]
Sorry, this doesn't ring a bell. Haven't used jsdom recently.

**Reply by sandersn** [reply]
The jsdom types on DT claim to be for jsdom@16. It imports ElementLocation from `@types/parse5@6`, but parse5@7 now ships its own types. The right thing to do is to update the DT jsdom types for jsdom@20 and make them depend on parse5 instead of `@types/parse5`.

Until then you have to choose between using jsdom@16 with types or jsdom@20 without types.

**Reply by lmeerma** [reply]
@sandersn Thank you.
>The right thing to do is to update the DT jsdom types for jsdom@20 and make them depend on parse5 instead of @types/parse5.

That was my thought as well, I was just looking for the maintainers of those types to make that happen. 😄

**Reply by sandersn** [reply]
Unfortunately, I think all four people you pinged are either [former] DT maintainers or have moved on to other things.

**Reply by SimenB** [reply]
Hit this in Jest, so made a start here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61299

**Reply by SimenB** [reply]
There is actually already #61170 and #61171, so hopefully movement soon 🙂

**Reply by SimenB** [reply]
Released, so this can be closed/resolved 👍
